quickwit-cluster/src/cluster.rs

```diff
@@ -326,9 +368,16 @@ fn create_failure_detector_config_for_test() -> FailureDetectorConfig {
 }
 
 /// Creates a local cluster listening on a random port.
-pub fn create_cluster_for_test(seeds: &[String]) -> anyhow::Result<Cluster> {
+pub fn create_cluster_for_test(seeds: &[String], services: &[&str]) -> anyhow::Result<Cluster> {
     let peer_uuid = Uuid::new_v4().to_string();
-    let cluster = create_cluster_for_test_with_id(peer_uuid, "test-cluster".to_string(), seeds)?;
+    let services = services
+        .iter()
+        .map(|service_str| QuickwitService::try_from(*service_str))
+        .collect::<Result<Vec<_>, _>>()?
+        .into_iter()
+        .collect::<HashSet<_>>();
```

Comment:
You can skip the `.collect::<Result<Vec<_>, _>>()?` above and directly collect into `.collect::<Result<HashSet<_>, _>>()`

Reply:
Unless my rust-analyser is fooling me again. This was not working. I will check.